packages/astro/src/runtime/server/render/component.ts

```diff
@@ -421,6 +421,9 @@ function renderAstroComponent(
 	slots: any = {}
 ): RenderInstance {
 	const instance = createAstroComponentInstance(result, displayName, Component, props, slots);
+	// Initialize the instance early so that child components with head propagation
+	// are able to register themselves.
+	instance.init(result);
```

Comment:
I am assuming this changes how long the server runtime waits before starting to stream <body>. It would be indented for cases where scripts and styles got missed, but could it change the timing for other users as well?

Reply:
Check out the new changes.